### PR TITLE
Bookkeeping of hydro-mc coupling in packet evolution

### PR DESCRIPTION
**Pull request by ffoucart**
## Proposed changes

Add computation of coupling between MC packet and fluid variables in EvolvePacketInElement. This is only bookkeeping for now, and not used to feedback on the fluid. We will also need to add contribution from the packets using the diffusion approximation after merging #5851 

### Upgrade instructions

None

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Evolution/Particles/MonteCarlo/Test_EvolvePackets.cpp
Comment (ffoucart) [inline tests/Unit/Evolution/Particles/MonteCarlo/Test_EvolvePackets.cpp]: I'll update the comment after code review -- but in fact, due to the use of a minimum value of 1.e-100 in the RNG when determining the time to the next event, we are explicitly guaranteed that no scattering / absorption will happen.

File: src/Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp
Comment (kidder) [inline src/Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp]: I made a request in #5903 to move this to the Units.hpp file; just want to make sure it doesn't get add here

File: src/Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp
Comment (kidder) [inline src/Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp]: Make this a doxygen comment with the first sentence as \brief and the rest as \details

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash changes, but will need to wait until #5903 is merged in order to rebase so that you can get the proton mass from the correct place.

**Comment by ffoucart** [comment]
Ok, rebased against develop, including #5903 .
I'm also realizing that I'll need to move the calculation of the coupling for the evolution along geodesics in a different file for #5927 to work... but I can deal with that in the relevant PR.

**Review by kidder — APPROVED** [review]

**Pull request merged**
